spec.md

```diff
@@ -849,6 +847,15 @@ error responses, see [Service Broker Errors](#service-broker-errors).
 }
 ```
 
+### Failure Cleanup
+
+After an unsuccessful provision attempt, a Service Broker MAY have stray
+resources persisted as a side effect. In order to prevent resource leaks, if a
```

Comment:
the use of MAY is funny here, this reads as the Service broker is allowed to have stray resources. Is this intended? If it is, then MAY is the correct wording.

Reply:
That is intended. From previous discussions on this topic, people were saying they wanted the option for Service Brokers to have broken remnants of resources left behind after a failure so that they can troubleshoot.

Reply:
Toughie... but, the MAY there is kind of awkward to read because, as scott said, it almost encourage people to leave stuff around.  I wonder if we could reword it a little to get the same net result:
```
After an unsuccessful provision attempt, the Service Broker might not be able to,
or might not want to, immediately delete any resources persisted as a side effect
of the provision request. In order to prevent...
```
This way we're not encouraging, or discouraging, it - we're just stating our reason
for why the Platform needs to send the deprovision.

Reply:
Hmm... Yeah, I'd also prefer if we didn't sound like we recommend it, so I'll get rid of the "MAY". The wording here's a bit tricky because it's also trying to encapsulate the timeout situation where the Service Broker doesn't realize yet the Platform considered the request a failure. I'd prefer keeping the motivation as general as possible.

In fact, it's not really "stray resources" we should care about; it's more generally just any side effects, so how about:

`There could potentially be side effects on the Service Broker as a result of an unsuccessful provision attempt. If a provision request has been sent...`